go-controller/pkg/node/gateway_udn.go

```diff
@@ -124,6 +127,14 @@ func (udng *UserDefinedNetworkGateway) addUDNManagementPort() error {
 			}
 		}
 	}
+
+	if err := util.UpdateNodeManagementPortMACAddresses(udng.node, udng.nodeAnnotator, macAddress, udng.GetNetworkName()); err != nil {
```

Comment:
don't we need to update `node-mgmt-port-mac-addresses` annotation when `deleteUDNManagementPort` is invoked for network deletion ?

Reply:
YEAHHHHHHHH UGHHH I need some integration tests for this PR

Reply:
I get this feeling this same bug might exist for other annotations that are network aware..? need to check

Reply:
The stale entry seems to exist only for `k8s.ovn.org/node-mgmt-port-mac-addresses`.

After network create:

```
    k8s.ovn.org/network-ids: '{"default":"0","tenantblue":"1"}'
    k8s.ovn.org/node-chassis-id: 4daa434a-e725-43b7-9157-bcf7d63c0d76
    k8s.ovn.org/node-gateway-router-lrp-ifaddrs: '{"default":{"ipv4":"100.64.0.3/16"},"tenantblue":{"ipv4":"100.65.0.3/16"}}'
    k8s.ovn.org/node-id: "3"
    k8s.ovn.org/node-mgmt-port-mac-addresses: '{"default":"f2:84:56:be:19:4f","tenantblue":"0a:4a:44:b3:5a:da"}'
```

After network delete:    

```
    k8s.ovn.org/network-ids: '{"default":"0"}'
    k8s.ovn.org/node-chassis-id: 4daa434a-e725-43b7-9157-bcf7d63c0d76
    k8s.ovn.org/node-gateway-router-lrp-ifaddrs: '{"default":{"ipv4":"100.64.0.3/16"}}'
    k8s.ovn.org/node-id: "3"
    k8s.ovn.org/node-mgmt-port-mac-addresses: '{"default":"f2:84:56:be:19:4f","tenantblue":"0a:4a:44:b3:5a:da"}'
```

Reply:
yeah you are right,, surprising other annotations look good haha, let me check